include/deal.II/base/exceptions.h

```diff
@@ -1163,6 +1163,36 @@ namespace StandardExceptions
                                            dealii::ExcCudaError(cudaGetErrorString(error_code)))
 #endif
 
+/**
+ * Exception for MPI errors. This exception should be used in an
+ * <code>AssertThrow</code> to check error codes of MPI functions. For
+ * example:
+ * @code
+ * const int ierr = MPI_Isend(...);
+ * AssertThrow(ierr == MPI_SUCCESS, ExcMPI(ierr));
+ * @endcode
+ *
+ * If the assertion fails then the error code will be used to print a helpful
+ * message to the screen by utilizing the <code>MPI_Error_string</code>
+ * function.
+ *
+ * This exception can be used when <code>deal.II</code> is compiled without
+ * MPI support: in that case it simply prints a message including the
+ * numerical value of the error code.
```

Comment:
In what circumstances would we check error code then? Or are you simply saying that one can use the exception without mpi (even though that would be in a context in which it is clearly not appropriate)?